pkg/secrethub/credentials/creators.go

```diff
@@ -31,13 +34,36 @@ func CreateKey() *KeyCreator {
 }
 
 // Create generates a new key and stores it in the KeyCreator.
-func (c *KeyCreator) Create() (Verifier, Encrypter, map[string]string, error) {
+func (kc *KeyCreator) Create() error {
 	key, err := GenerateRSACredential(crypto.RSAKeyLength)
 	if err != nil {
-		return nil, nil, nil, err
+		return err
 	}
-	c.key = key
-	return c.key, c.key, map[string]string{}, nil
+	kc.key = key
+	return nil
+}
+
+func (kc *KeyCreator) Metadata() map[string]string {
+	return map[string]string{}
+}
+
+type awsCreator struct {
```

Comment:
This type is stuck in between multiple functions of the `KeyCreator`. Let's move  it somehwere else in the file for clarity.

Reply:
Is it though? I moved the `CreateAWS()` function above this. Is it clearer now?

Reply:
AH cool, maybe I misread. Looks good. 👍